spyder/utils/external/lockfile.py

```diff
@@ -182,10 +182,12 @@ def lock(self):
                             conditions = ['spyder' in c.lower()
                                           for c in p.cmdline()]
                         else:
-                            conditions = [p.name() == 'spyder',
-                                          p.name() == 'spyder3']
-                        # For DEV
-                        conditions += ['bootstrap.py' in p.cmdline()]
+                            # Valid names for main script
+                            names = set(['spyder', 'spyder3', 'bootstrap.py'])
+                            # Check the first three command line arguments
+                            arguments = set(os.path.basename(arg)
+                                            for arg in p.cmdline()[:3])
+                            conditions = [names & arguments]

```

Comment:
What does this line do?

Reply:
First gets the basename for each of the command line arguments, up to the first 3. For example, ```['/Users/gkv/anaconda/python.app/Contents/MacOS/python', '/Users/gkv/anaconda/bin/spyder']``` would become ```set(['python', 'spyder'])```, then it does an intersection of the sets so only the values in both sets remain.

Reply:
So why do you put the intersection in a list and then use `if not any`? 

How about `if not (names & arguments)` ?

Reply:
I kept the existing convention. Otherwise It would require changing the whole if block. From an efficiency standpoint, there is little to no gain.